cmd/managed.go

```
@@ -0,0 +1,63 @@
+package cmd
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/twpayne/chezmoi/internal/chezmoi"
+)
+
+var managedCmd = &cobra.Command{
+	Use:     "managed",
+	Args:    cobra.NoArgs,
+	Short:   "List the managed files in the destination directory",
+	Long:    mustGetLongHelp("managed"),
+	Example: getExample("managed"),
+	PreRunE: config.ensureNoError,
+	RunE:    config.runManagedCmd,
+}
+
+func init() {
+	rootCmd.AddCommand(managedCmd)
+}
+
+func recurseEntries(in []chezmoi.Entry) []string {
+	out := make([]string, len(in))
+	for _, entry := range in {
+		switch v := entry.(type) {
+		case *chezmoi.Dir:
+			entries := make([]chezmoi.Entry, len(v.Entries))
+			for _, entry := range v.Entries {
+				entries = append(entries, entry)
+			}
+			results := recurseEntries(entries)
+			out = append(out, results...)
+		case *chezmoi.File, *chezmoi.Symlink:
+			out = append(out, v.TargetName())
+		}
+	}
+	return out
+}
+
+func (c *Config) runManagedCmd(cmd *cobra.Command, args []string) error {
+	ts, err := c.getTargetState(nil)
+	if err != nil {
+		return err
+	}
+	entries := make([]chezmoi.Entry, len(ts.Entries))
+	for _, entry := range ts.Entries {
+		entries = append(entries, entry)
+	}
+	allManaged := recurseEntries(entries)
+	for _, tn := range allManaged {
+		path := filepath.Join(ts.DestDir, tn)
+		entry, _ := ts.Get(c.fs, path)
```

This should return any error, rather than discarding it.